File: search/binary.js
function sorting(arr, n) {
    for (let i = 0; i < n - 1; i++) {
        let minSize = i;
        for (let j = i + 1; j < n; j++) {
            if (arr[j] < arr[minSize]) {
                minSize = j;
            }
        }
        let temp = arr[i];
        arr[i] = arr[minSize];
        arr[minSize] = temp;
    }
}

function binary(arr, l, r, target) {
    while (l <= r) {
        let m = l + Math.floor((r - l) / 2);
        if (arr[m] === target) {
            return true;
        } else if (arr[m] < target) {
            l = m + 1;
        } else {
            r = m - 1;
        }
    }
    return false;
}

const MAX_SIZE = 100;
let arr = new Array(MAX_SIZE);
let n, i;
let searchQuery;

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter the number of elements (up to " + MAX_SIZE + "): ", (answer) => {
    n = parseInt(answer);
    console.log("Enter the elements:");
    rl.on('line', (line) => {
        arr[i] = parseInt(line);
        i++;
        if (i === n) {
            rl.close();
            processInput();
        }
    });
});

function processInput() {
    console.log("The elements entered are: " + arr.join(' '));
    rl.question("Enter the element to be searched: ", (answer) => {
        searchQuery = parseInt(answer);
        sorting(arr, n);
        if (binary(arr, 0, n - 1, searchQuery)) {
            console.log("The element exists.");
        } else {
            console.log("The element doesn't exist in the array.");
        }
    });
}

File: search/binary.py
def sorting(arr, n):
    for i in range(n - 1):
        minSize = i
        for j in range(i + 1, n):
            if arr[j] < arr[minSize]:
                minSize = j
        arr[i], arr[minSize] = arr[minSize], arr[i]

def binary(arr, l, r, target):
    while l <= r:
        m = l + (r - l) // 2
        if arr[m] == target:
            return True
        elif arr[m] < target:
            l = m + 1
        else:
            r = m - 1
    return False

MAX_SIZE = 100
arr = [0] * MAX_SIZE
n = 0
i = 0
searchQuery = 0

n = int(input("Enter the number of elements (up to {}): ".format(MAX_SIZE)))
print("Enter the elements:")
while i < n:
    arr[i] = int(input())
    i += 1

print("The elements entered are:", end=" ")
print(*arr[:n])

searchQuery = int(input("Enter the element to be searched: "))

sorting(arr, n)

if binary(arr, 0, n - 1, searchQuery):
    print("The element exists.")
else:
    print("The element doesn't exist in the array.")
